### PR TITLE
Add tree-sitter-eex parser for (L)EEx files

### PR DESCRIPTION
**Pull request by connorlay**
Related to https://github.com/nvim-treesitter/nvim-treesitter/issues/1352

This PR adds support for EEx and LEEx files with [tree-sitter-eex](https://github.com/connorlay/tree-sitter-eex). Unlike HEEx, EEx does not make any assumptions about what language it is injecting Elixir into. The addition of this parser to nvim-treesitter should improve the experience of using Neovim and Tree Sitter on older Phoenix applications that have not migrated to HEEx yet, as well as other usages of EEx that do not target HTML.

### CODE REVIEW

File: ftdetect/eex.vim [hunk 1]
```diff
@@ -0,0 +1,2 @@
+au BufRead,BufNewFile *.eex set filetype=eex
```
Comment (clason) [inline ftdetect/eex.vim:1]: We're trying to get rid of `ftdetect` rules in this plugin... #2259

Comment (connorlay) [reply ftdetect/eex.vim:1]: Ah ok, according to this comment I need to contribute upstream to Vim, then have that come to Neovim via a patch?

https://github.com/nvim-treesitter/nvim-treesitter/pull/2268#issuecomment-1013878699

Comment (clason) [reply ftdetect/eex.vim:1]: That would be ideal, yes. (Such a trivial change usually gets merged quickly.)

Comment (clason) [reply ftdetect/eex.vim:1]: In any case, if you drop that file from the PR, we could merge it now. (You can tell your users to put the autocommand in their `init.vim` in the meantime.)

Comment (connorlay) [reply ftdetect/eex.vim]: @clason I removed the `ftdetect` rules as requested. Is there anything else you'd like to see in this PR?

**Review by clason — APPROVED** [review]

**Pull request merged**